src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,340 @@
+
+    private readonly List<TweakedColorPickerPreset> presets = new List<TweakedColorPickerPreset>();
+
+    private ToolButton pickerButton;
+    private CheckButton hsvCheckButton;
+    private CheckButton rawCheckButton;
+    private LineEdit hexColorEdit;
+    private HSeparator separator;
+    private GridContainer presetsContainer;
+    private Button addPresetButton;
+
+    private bool hsvButtonEnabled = true;
+    private bool rawButtonEnabled = true;
+    private bool presetsEnabled = true;
+    private bool presetsVisible = true;
+
+    /// <summary>
+    ///   Decide if user can toggle HSV CheckButton to switch HSV mode.
+    /// </summary>
+    [Export]
+    public bool HSVButtonEnabled
+    {
+        get => hsvButtonEnabled;
+        set
+        {
+            hsvButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    /// <summary>
+    ///   Decide if user can toggle Raw CheckButton to switch Raw mode.
+    /// </summary>
+    [Export]
+    public bool RawButtonEnabled
+    {
+        get => rawButtonEnabled;
+        set
+        {
+            rawButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool HsvMode
+    {
+        get => base.HsvMode;
+        set
+        {
+            if (value && RawMode)
+                return;
+
+            base.HsvMode = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool RawMode
+    {
+        get => base.RawMode;
+        set
+        {
+            if (value && HsvMode)
+                return;
+
+            base.RawMode = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool PresetsEnabled
+    {
+        get => presetsEnabled;
+        set
+        {
+            presetsEnabled = value;
+
+            if (addPresetButton == null)
+                return;
+
+            addPresetButton.Disabled = !value;
+        }
+    }
+
+    [Export]
+    public new bool PresetsVisible
+    {
+        get => presetsVisible;
+        set
+        {
+            presetsVisible = value;
+
+            if (presetsContainer == null)
+                return;
+
+            separator.Visible = value;
+            presetsContainer.Visible = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        // Hide replaced native controls. Can't delete them because it will crash Godot.
+        GetChild(4).GetChild<Control>(4).Hide();
+        GetChild<Control>(5).Hide();
+        GetChild<Control>(6).Hide();
+        GetChild<Control>(7).Hide();
+
+        pickerButton = GetChild(1).GetChild<ToolButton>(1);
+        hsvCheckButton = GetNode<CheckButton>("ButtonsContainer/HSVCheckButton");
+        rawCheckButton = GetNode<CheckButton>("ButtonsContainer/RawCheckButton");
+        hexColorEdit = GetNode<LineEdit>("ButtonsContainer/HexColorEdit");
+        hexColorEdit.Text = Color.ToHtml();
```

Comment:
Use the already existing `OnColorChanged` method and pass in the current color. And this should be moved somewhere in the bottom as this particular section is for node retrievals.